### PR TITLE
Add an E2E test for apparmor profile recording and fix various issues related to this

### PR DESCRIPTION
**Pull request by ccojocar**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add an E2E test for apparmor profile recording and fix various issues related to in-cluster apparmor profile recording
- **Extend the Debian vagrant file to install and setup a 1 node k8s cluster**
- **Name the baseprofile test consitently with other E2E tests**
- **Refactor baseprofiles e2e script**
- **Add an e2e test to record apparmor profiles**
- **Enable in CI the e2e test for apparmor profile recording**

#### Does this PR have test?

Yes

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

cc @mhils

### CODE REVIEW

**Comment by ccojocar** [comment]
@saschagrunert any idea what's causing this nix issue here https://github.com/kubernetes-sigs/security-profiles-operator/actions/runs/9880101347/job/27287874214? 

The same Debian 12 VM boots totally fine in this test https://github.com/kubernetes-sigs/security-profiles-operator/actions/runs/9880101347/job/27288033488.

**Comment by saschagrunert** [comment]
@ccojocar that was something unfixed in the past. Can you re-use the image from the `image` build step? https://github.com/kubernetes-sigs/security-profiles-operator/actions/runs/9880101347/job/27287874725

**Comment by saschagrunert** [comment]
@ccojocar yeah we had this issue in the past.

But can we just re-use the image from the `image` build step instead of building it again?

**Comment by ccojocar** [comment]
> But can we just re-use the image from the image build step instead of building it again?

Thanks for pointing this out. Now I realized the difference.  I think the vagrant file for Debian which was used by e2e-spoc test was a bit lighter, and didn't include the image.

**Comment by saschagrunert** [comment]
> > But can we just re-use the image from the image build step instead of building it again?
> 
> Thanks for pointing this out. Now I realized the difference. I think the vagrant file for Debian which was used by e2e-spoc test was a bit lighter, and didn't include the image.

If we don't need the image than we may just pre-build the spoc binary or such.

File: hack/ci/Vagrantfile-debian [hunk 1]
```diff
@@ -53,6 +54,37 @@ Vagrant.configure("2") do |config|
...
       /vagrant/hack/install-libbpf.sh
 
+      # Install cosign (required by e2e-baseprofile test)
+      COSIGN_VERSION=v2.2.1
+      COSIGN_BINARY=/usr/bin/cosign
+      curl -sSfL --retry 5 --retry-delay 3 "https://github.com/sigstore/cosign/releases/download/$COSIGN_VERSION/cosign-linux-amd64" -o "$COSIGN_BINARY"
+      chmod +x "$COSIGN_BINARY"
+
+      # Install kubernetes related binaries
+      KUBERNETES_VERSION=v1.30
```
Comment (saschagrunert) [inline hack/ci/Vagrantfile-debian:64]: Are those versions part of dependencies.yaml?

File: hack/ci/e2e-seccomp.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+# Copyright 2022 The Kubernetes Authors.
```
Comment (saschagrunert) [inline hack/ci/e2e-seccomp.sh:2]:
```suggestion
# Copyright 2024 The Kubernetes Authors.
```

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**